azure/durable_functions/models/DurableEntityContext.py

```diff
@@ -0,0 +1,219 @@
+from typing import Optional, Any, Dict, Tuple, List, Callable
+from azure.functions._durable_functions import _deserialize_custom_object
+from datetime import datetime
+import json
+
+
+class DurableEntityContext:
+    """Context of the durable entity context.
+
+    Describes the API used to specify durable entity user code.
+    """
+
+    def __init__(self,
+                 name: str,
+                 key: str,
+                 exists: bool,
+                 state: Any):
+        """Context of the durable entity context.
+
+        Describes the API used to specify durable entity user code.
+
+        Parameters
+        ----------
+        name: str
+            The name of the Durable Entity
+        key: str
+            The key of the Durable Entity
+        exists: bool
+            Flag to determine if the entity exists
+        state: Any
+            The internal state of the Durable Entity
+        """
+        self._entity_name: str = name
+        self._entity_key: str = key
+
+        self._exists: bool = exists
+        self._is_newly_constructed: bool = False
+
+        self._state: Any = state
+        self._input: Any = None
+        self._operation: Optional[str] = None
+        self._result: Any = None
+
+    @property
+    def entity_name(self) -> str:
+        """Get the name of the Entity.
+
+        Returns
+        -------
+        str
+            The name of the entity
+        """
+        return self._entity_name
+
+    @property
+    def entity_key(self) -> str:
+        """Get the Entity key.
+
+        Returns
+        -------
+        str
+            The entity key
+        """
+        return self._entity_key
+
+    @property
+    def operation_name(self) -> Optional[str]:
+        """Get the current operation name.
+
+        Returns
+        -------
+        Optional[str]
+            The current operation name
+        """
+        # TODO: Maybe we should raise an
+        # exception if _operation is None
+        return self._operation
+
+    @property
+    def is_newly_constructed(self) -> bool:
+        """Determine if the Entity was newly constructed.
+
+        Returns
+        -------
+        bool
+            True if the Entity was newly constructed. False otherwise.
+        """
+        # TODO: not updating this atm
```

Comment:
Not sure if this even gets updated when C# handles a batch either, so this may not be necessary. We should consult Sebastian about this.

Reply:
@sebastianburckhardt:

Is there a  use for a `isNewlyConstructed` flag for entities? Or can it be safely removed?

Reply:
I'll be tracking this here: https://github.com/Azure/azure-functions-durable-python/issues/222